File: Day 9/3 Max Sub Array/solution.py
   def maxSubArray(self, nums: List[int]) -> int:
        maxSum = nums[0]
        currentMax = 0
        for i in range(len(nums)):
            if currentMax < 0:
                currentMax = nums[i]
            else:
                currentMax = nums[i] + currentMax
            if currentMax > maxSum:
                maxSum = currentMax
        return maxSum

File: Day 9/3 Max Sub Array/solution.js
function maxSubArray(nums) {
    let maxSum = nums[0];
    let currentMax = 0;
    for (let i = 0; i < nums.length; i++) {
        if (currentMax < 0) {
            currentMax = nums[i];
        } else {
            currentMax = nums[i] + currentMax;
        }
        if (currentMax > maxSum) {
            maxSum = currentMax;
        }
    }
    return maxSum;
}